lib/ngx/process.lua

```diff
@@ -0,0 +1,49 @@
+-- Copyright (C) Yichun Zhang (agentzh)
+
+
+local ffi = require 'ffi'
+local base = require "resty.core.base"
+local getfenv = getfenv
+
+local process_types = {
+    [base.FFI_PROCESS_SINGLE    ]  = "single process",
+    [base.FFI_PROCESS_MASTER    ]  = "master process",
+    [base.FFI_PROCESS_SIGNALLER ]  = "signaller process",
+    [base.FFI_PROCESS_WORKER    ]  = "worker process",
+    [base.FFI_PROCESS_HELPER    ]  = "helper process",
+    [base.FFI_PROCESS_PRIVILEGED]  = "privileged agent process",
+}
+
+
+local C = ffi.C
+local _M = { version = base.version }
+
+
+ffi.cdef[[
+void ngx_http_lua_ffi_privileged_agent(int enable);
+int ngx_http_lua_ffi_process_type(void);
```

Comment:
Again, we should use verbs in the function names. `ngx_http_lua_ffi_get_process_type` is a better name.